### PR TITLE
Fixes prediction for rigid-batch models

### PR DESCRIPTION
**Pull request by rbharath**
Will add details once tests pass.

### CODE REVIEW

**Comment by rbharath** [comment]
This PR fixes prediction for those tensorflow models which have rigid batch-size requirements. It does so by adding a `pad_batch` flag to `predict_on_batch` and `predict_proba_on_batch`

Also, contains some other miscellaneous bugfixes (better handling of 1-d featurizers, and removal of a double-softmax for tensorflow classifier models).
